purepyvmware/datastores.py

```diff
@@ -0,0 +1,329 @@
+
+        self.rescan_esxi_storage(connected_esxi_hosts)
+        storage_manager = self.sms_instance.QueryStorageManager()
+        storage_containers = storage_manager.QueryStorageContainer().storageContainer
+
+        for container in storage_containers:
+            # Example of arrayId output:
+            # (str) ['com.purestorage:3b7b308d-98f9-425e-87a1-3e57ada49658']
+            if container.arrayId[0].split(':')[1] == flasharray.get()['id']:
+                sc_id = container.uuid
+
+        for esxi_host in connected_esxi_hosts:
+            host_dssystem = esxi_host.configManager.datastoreSystem
+            vvol_spec = vim.HostDatastoreSystemVvolDatastoreSpec(name=ds_name, scId=sc_id)
+            vvol_datastore = host_dssystem.CreateVvolDatastore(vvol_spec)
+
+        self.rescan_esxi_storage(connected_esxi_hosts)
+
+        return vvol_datastore
+
+    def get_pure_provider(self, storage_manager, registration_url):
+        """Get specified Pure Storage Provider from Storage Monitoring Services (SMS) instance.
+
+        Args:
+            storage_manager (sms.StorageManager): SMS storage manager from authenticated instance.
+            registration_url (str): URL that the storage provider was registered with.
+
+        Returns:
+            matched_provider (sms.provider.VasaProvider): Requested VASA Provider object from SMS.
+
+        Raises:
+            ValueError:
+                - If specified provider is not found on SMS instance.
+        """
+        matched_provider = None
+        providers = storage_manager.QueryProvider()
+
+        for provider in providers:
+            # Example URL: https://10.10.10.10:8084/version.xml
+            if provider.QueryProviderInfo().url == registration_url:
+                matched_provider = provider
+                break
+
+        if not matched_provider:
+            raise ValueError('Registered provider not found. Verify the registration URL and try again.')
+
+        return matched_provider
+
+    def register_storage_provider(self, flasharray, ip_address):
+        """Register a Pure Storage FlashArray as a storage provider with vCenter Server.
+
+        Args:
+            flasharray (purestorage.FlashArray): Verified or unverified session to FlashArray.
+            ip_address (str): IP address of the controller you want to register with vCenter (not the 'vir0/1' address)
+
+        Returns:
+            new_provider (sms.provider.VasaProvider): Newly created VASA Provider object from SMS.
+
+        Raises:
+            ValueError:
+                - If FlashArray controller is not found as correlated with the supplied IP address.
+        """
+        all_networks = flasharray.list_network_interfaces()
+        fa_ctlr = None
+        fa_name = flasharray.get()['array_name']
+        storage_manager = self.sms_instance.QueryStorageManager()
+
+        for network in all_networks:
+            if ('vir' not in network['name']) and network['address'] == ip_address:
```

Comment:
nit: the `()` are not needed here. With this being an `and` statement the whole thing must evaluate to True.

If you want to keep it for readability, that's fine, but just know it's not going to change how this line evaluates.